Framework/API/src/LogManager.cpp

```diff
@@ -662,9 +663,13 @@ Kernel::TimeSeriesProperty<bool> *LogManager::getInvalidValuesFilter(const std::
   return nullptr;
 }
 
-bool LogManager::operator==(const LogManager &other) const { return *m_manager == *(other.m_manager); }
+bool LogManager::operator==(const LogManager &other) const {
+  return (*m_manager == *(other.m_manager)) && (*m_timeroi == *(other.m_timeroi));
```

Comment:
why are we including equality for `m_timeroi`?
The current `LogManager::operator== checks` for [equality between corresponding properties](https://github.com/mantidproject/mantid/blob/main/Framework/Kernel/src/PropertyManager.cpp#L604) for the two property managers, but [TimeSeriesProperty::operator==](https://github.com/mantidproject/mantid/blob/main/Framework/Kernel/src/TimeSeriesProperty.cpp#L172) does _not_ check for equality between the filters of the properties ([TimeSeriesProperty::m_filter](https://github.com/mantidproject/mantid/blob/main/Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h#L376)). This check for filter equality would be the closest thing to the check you introduce for equality between the `LogManager::m_timeroi` values

Reply:
You are asking a very good question. If we add the `TimeROI` comparison here, it should be in `FilteredTimeSeriesProperty` as well.